File: src/static/js/business_plan.js
```javascript
// Business Plan JavaScript Functions

// Function to create a new business plan
function createPlan() {
    let planData = {
        title: document.getElementById('plan-title').value,
        description: document.getElementById('plan-description').value,
        industry: document.getElementById('plan-industry').value,
        marketAnalysis: document.getElementById('plan-market-analysis').value,
        financialPlan: document.getElementById('plan-financial-plan').value,
    };

    fetch('/api/business_plan', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(planData),
    })
    .then(response => response.json())
    .then(data => {
        if (data.success) {
            alert('Business plan created successfully!');
            location.reload();
        } else {
            alert('Error: ' + data.message);
        }
    });
}

// Function to edit an existing business plan
function editPlan(planId) {
    let planData = {
        title: document.getElementById('plan-title').value,
        description: document.getElementById('plan-description').value,
        industry: document.getElementById('plan-industry').value,
        marketAnalysis: document.getElementById('plan-market-analysis').value,
        financialPlan: document.getElementById('plan-financial-plan').value,
    };

    fetch('/api/business_plan/' + planId, {
        method: 'PUT',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(planData),
    })
    .then(response => response.json())
    .then(data => {
        if (data.success) {
            alert('Business plan updated successfully!');
            location.reload();
        } else {
            alert('Error: ' + data.message);
        }
    });
}

// Function to delete a business plan
function deletePlan(planId) {
    fetch('/api/business_plan/' + planId, {
        method: 'DELETE',
    })
    .then(response => response.json())
    .then(data => {
        if (data.success) {
            alert('Business plan deleted successfully!');
            location.reload();
        } else {
            alert('Error: ' + data.message);
        }
    });
}
```

File: src/controllers/business_plan.py
```python
from flask import Blueprint, request, jsonify
from src.services.business_plan_service import create_plan, edit_plan, delete_plan

business_plan = Blueprint('business_plan', __name__)

@business_plan.route('/create', methods=['POST'])
def create():
    data = request.get_json()
    result = create_plan(data)
    if result:
        return jsonify({"message": "plan_saved", "data": result}), 201
    else:
        return jsonify({"message": "Error in saving plan"}), 400

@business_plan.route('/edit/<plan_id>', methods=['PUT'])
def edit(plan_id):
    data = request.get_json()
    result = edit_plan(plan_id, data)
    if result:
        return jsonify({"message": "plan_saved", "data": result}), 200
    else:
        return jsonify({"message": "Error in editing plan"}), 400

@business_plan.route('/delete/<plan_id>', methods=['DELETE'])
def delete(plan_id):
    result = delete_plan(plan_id)
    if result:
        return jsonify({"message": "plan_deleted"}), 200
    else:
        return jsonify({"message": "Error in deleting plan"}), 400
```